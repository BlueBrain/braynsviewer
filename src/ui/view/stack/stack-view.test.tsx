// To test a React component, you need to install these modules:
// yarn add --dev react-test-renderer @types/react-test-renderer
// @see https://jestjs.io/docs/en/snapshot-testing
//
// If a test failed just because you intended to improve the component,
// just call `jest --updateSnapshot`.

import React from "react"
import Renderer from "react-test-renderer"
import StackView, { StackViewProps } from "./stack-view"

function view(partialProps: Partial<StackViewProps>) {
    const props: StackViewProps = {
        children: [],
        value: "Hello",
        ...partialProps,
    }
    return Renderer.create(<StackView {...props} />).toJSON()
}

describe("<StackView/> in ui/view", () => {
    it("should be consistent with previous snapshot", () => {
        expect(view({})).toMatchSnapshot()
    })
})
