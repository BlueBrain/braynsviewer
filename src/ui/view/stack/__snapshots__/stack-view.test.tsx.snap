// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StackView/> in ui/view should be consistent with previous snapshot 1`] = `
<div
  className="custom ui-view-StackView"
/>
`;
